rts/Sim/Units/Unit.h

```diff
@@ -204,11 +207,15 @@ class CUnit : public CSolidObject
 	bool ScriptDecloak(const CSolidObject* object, const CWeapon* weapon);
 	bool GetNewCloakState(bool checkStun);
 
-	enum ChangeType {
-		ChangeGiven,
-		ChangeCaptured
+	// C++-side reasons for changing team.
+	// For the full list of reasons, see 'GG.CHANGETEAM_REASON' in luarules/gadgets.lua
```

Comment:
Engine generally shouldn't refer to Lua (even if it provides examples of its own). If Lua gets to define something then it's none of engine's business, and if engine has to define it then that is the source of truth and it is Lua who should refer to the engine.

Reply:
Hmm, yeah. It is just a pass through, but the engine does refer to this enum a few places and the values match. But your point is taken.

Reply:
The enum should be kept as-is, AFAICT its only use is to pick which of the two irrelevant stats this affects, "units sent" or "units outcaptured". These stats are a larger topic, see #2041 but the tl;dr is that we can't add a 3rd stat anyway for protocol reasons, and they would ideally be moved to Lua sometime anyway.
https://github.com/beyond-all-reason/RecoilEngine/blob/a9bd9de5ccd59cc28c13105f2d3c1249ae3c4f63/rts/Sim/Misc/Team.cpp#L440-L445

Lua doesn't need to have the reason passed via the callin, see https://github.com/beyond-all-reason/RecoilEngine/pull/2423#discussion_r2230781784